### ISSUE TITLE
Allow pest to match against byte literals

### ISSUE THREAD

**Issue by jstnlef** [pest-parser/pest#244]
For example, if we would like to match against an occurrence of one or more byte literals (like 7c or FFFF) we should be able to. This would take a bit of design work, however, since we make assumptions in many places in the code that we'll be dealing with UTF-8 strings.

Request from @Restioson

**Comment by dragostis** [comment]
The redesign should tackle the issue of capturing strings from the input with the generated `Span`s. Currently, these guarantee UTF-8 cheap captures.

**Comment by Restioson** [comment]
Yes hello, thanks. For more info: my usecase is matching actual *bytes* (not literals) for capturing values in AML.

**Comment by Restioson** [comment]
TBH we're probably going to do it with a handwritten parser @IsaacWoods wrote a while back (but nevertheless I really like this library and would like to keep helping). So, no timeframe really.

**Comment by iptq** [comment]
bump, any progress on this? i might be needing this soon (am parsing IMAP with pest) so i'll probably implement a proof-of-concept in a bit

**Comment by 12089897411** [comment]
Can you put it on the agenda?
